qp_klp/Step.py

```diff
@@ -369,32 +379,141 @@ def generate_sifs(self, from_qiita):
     def get_prep_file_paths(self):
         return self.prep_file_paths
 
-    def map_sample_names_to_tube_ids(self, prep_info_file_paths,
-                                     sn_tid_map_by_proj):
-        for proj in sn_tid_map_by_proj:
-            if sn_tid_map_by_proj[proj] is not None:
-                # this project has tube-ids registered in Qiita.
-                # find the prep-file associated with this project.
-                for prep_file in prep_info_file_paths:
-                    # not the best check but good enough for now.
-                    if proj in prep_file:
-                        df = pd.read_csv(prep_file, sep='\t',
-                                         dtype=str, index_col=False)
-                        # save copy of sample_name column as 'old_sample_name'
-                        df['old_sample_name'] = df['sample_name']
-                        for i in df.index:
-                            smpl_name = df.at[i, "sample_name"]
-                            if not smpl_name.startswith('BLANK'):
-                                # remove any leading zeroes if they exist
-                                smpl_name = smpl_name.lstrip('0')
-                                if smpl_name in sn_tid_map_by_proj[proj]:
-                                    tube_id = sn_tid_map_by_proj[proj][
-                                        smpl_name]
-                                    df.at[i, "sample_name"] = tube_id
-                        df.to_csv(prep_file, index=False, sep="\t")
-
-    def update_blanks_in_qiita(self, sifs, qclient):
-        for sif_path in sifs:
+    def get_tube_ids_from_qiita(self, qclient):
+        # Update get_project_info() so that it can return a list of
+        # samples in projects['samples']. Include blanks in projects['blanks']
+        # just in case there are duplicate qiita_ids
+        qiita_ids = list(set([x['qiita_id'] for x in
```

Comment:
Why not create the set from the beginning? 
qiita_ids = list({x['qiita_id'] for x in ... })

Reply:
It was an insurance measure for uniqueness during development, in case multiple projects with the same qiita-id were declared in a sample-sheet or pre-prep file. I removed the list() and set() casts entirely.